lightning/src/routing/gossip.rs

```diff
@@ -1578,17 +1578,33 @@ impl<L: Deref> NetworkGraph<L> where L::Target: Logger {
 		self.add_channel_between_nodes(msg.short_channel_id, chan_info, utxo_value)
 	}
 
+	#[cfg(feature = "std")]
 	/// Marks a channel in the graph as failed if a corresponding HTLC fail was sent.
 	/// If permanent, removes a channel from the local storage.
 	/// May cause the removal of nodes too, if this was their last channel.
 	/// If not permanent, makes channels unavailable for routing.
+	///
+	/// This method is only available with the `std` feature. See
+	/// [`NetworkGraph::channel_failed_with_time`] for `no-std` use.
 	pub fn channel_failed(&self, short_channel_id: u64, is_permanent: bool) {
```

Comment:
This is called in the `NetworkGraph`'s event handler, which doesn't have any conditioning on `std`.

Reply:
Ahh right. I think we run into the issue of not being able to pass through current_time_unix from somewhere for no-std. I’ll think about it for a bit.

Reply:
So there's no way for us to pass through current UNIX time in the even handler for no-std without changing the `EventHandler` and `EventsProvider` traits to maybe support passing it in form the public API :/

Reply:
Where would someone using `no-std` get the current Unix time from? If they could implement `util::time::Time` to return it, then `NetworkGraph` could be parameterized by it if we desire this behavior in `no-std`.

Reply:
No worries. I kind of didn't see the whole picture until going back and forth on this one.